### PR TITLE
enable julia arg `--compiled-modules` during testing

### PR DESCRIPTION
**Pull request by haberdashPI**
Let's you specify whether modules should be compiled during tests.

c.f. https://github.com/beacon-biosignals/StableHashTraits.jl/issues/40

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -32,6 +32,9 @@ inputs:
   annotate:
     description: 'Whether or not to attempt to create GitHub annotations to show test failures inline. Only effective on Julia 1.8+.'
     default: 'false'
+  compiled_modules:
+    description: 'Whether to run tests with `compiled-modules` equal to `yes` or `no`'
+    default: 'yes'
```
Comment (IanButterworth) [inline action.yml:37]: Julia can also take "existing" as an option

**Comment by haberdashPI** [comment]
Bump! Just checking if there's anything I can to do get this merged.

File: action.yml (hunk 1 above)
Comment (haberdashPI) [reply action.yml:37]: Weirdly this is not documented in julia --help 🤔 Is that an oversight.

Comment (IanButterworth) [reply action.yml]: It's on 1.11-DEV https://docs.julialang.org/en/v1.11-dev/manual/command-line-interface/#command-line-interface

Comment (IanButterworth) [reply action.yml]: Perhaps just don't specify the accepted values and point users to the julia docs?

Comment (haberdashPI) [reply action.yml]: Ah, that makes sense. Okay!

**Review by IanButterworth — APPROVED** [review]

**Comment by IanButterworth** [comment]
Once this is merged can you give this a try in the wild using the `julia-actions/julia-runtest@main` tag and check that it works, then I'll make a release

**Review by haberdashPI — COMMENTED** [review]
> Once this is merged can you give this a try in the wild using the `julia-actions/julia-runtest@main` tag and check that it works, then I'll make a release 

Sounds perfect! I'm going to make a PR to address beacon-biosignals/StableHashTraits.jl#40, and I'll see how it goes there.

**Pull request merged**